docs/e2e.tf

```diff
@@ -0,0 +1,579 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "=4.14.0"
+    }
+  }
+}
+
+# If using the "az" azure cli for authentication,
+# you will need `ARM_SUBSCRIPTION_ID` in the environment - see:
+# https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/guides/azure_cli
+provider "azurerm" {
+  features {}
+}
+
+variable "ssh_key" {
+  type        = string
+  description = "public ssh key to be added to all created VMs (for the 'ubuntu' user)"
+}
+
+resource "azurerm_resource_group" "tailscale_testing" {
+  name     = "tailscale"
+  location = "Australia East"
+}
+
+resource "azurerm_virtual_network" "vnet" {
+  name                = "tailscale-vnet"
+  address_space       = ["10.0.0.0/16"]
+  location            = azurerm_resource_group.tailscale_testing.location
+  resource_group_name = azurerm_resource_group.tailscale_testing.name
+}
+
+resource "azurerm_virtual_network" "vnet2" {
+  name                = "tailscale-vnet2"
+  address_space       = ["10.1.0.0/16"]
+  location            = azurerm_resource_group.tailscale_testing.location
+  resource_group_name = azurerm_resource_group.tailscale_testing.name
+}
+
+resource "azurerm_subnet" "vnet_subnet_1" {
+  name                 = "tailscale-vnet_subnet_1"
+  resource_group_name  = azurerm_resource_group.tailscale_testing.name
+  virtual_network_name = azurerm_virtual_network.vnet.name
+  address_prefixes     = ["10.0.1.0/24"]
+}
+
+resource "azurerm_subnet" "vnet_subnet_2" {
+  name                 = "tailscale-vnet_subnet_2"
+  resource_group_name  = azurerm_resource_group.tailscale_testing.name
+  virtual_network_name = azurerm_virtual_network.vnet2.name
+  address_prefixes     = ["10.1.2.0/24"]
+}
+
+# TODO: these domain names are public (derper and headscale) so there is chance of conflict if two people deploy this testing simultaneously or simply the domain name is taken - maybe these should be variables for this document and terraform?
```

Comment:
This is something I'd like to create an issue for and address in a later followup. It feels low priority and it will take some time to tweak the document to work with these variables.